src/pudl/workspace/setup.py

```diff
@@ -10,6 +13,55 @@
 logger = pudl.logging_helpers.get_logger(__name__)
 
 
+def get_settings(
+    yaml_file: IO | None, tmp_dir: str | None = None, live_dbs: bool = False
+) -> dict[str, str]:
+    """Handle PUDL settings and env vars."""
+    if yaml_file is not None:
+        settings = yaml.safe_load(yaml_file)
+    else:
+        settings = {}
+
+    if (pudl_out := os.getenv("PUDL_OUTPUT")) and (pudl_in := os.getenv("PUDL_INPUT")):
+        derive_paths_in = Path(pudl_in).parent
+        derive_paths_out = Path(pudl_out).parent
+    elif settings:
+        derive_paths_in = settings["pudl_in"]
+        derive_paths_out = settings["pudl_out"]
+    else:
+        raise RuntimeError(
+            "Must set 'PUDL_OUTPUT'/'PUDL_INPUT' environment variables or provide valid yaml config file."
+        )
+
+    if tmp_dir and not live_dbs:
+        derive_paths_out = tmp_dir
```

Comment:
I think we'll also have to handle the `--tmp-data` case - which is actually what motivated me to think about passing in `input_dir` and `output_dir`.

Then, this settings function doesn't need to know about all the weird CLI options, and we can have the fixtures "simply" pass in the right in/out dirs based on the CLI flags.